Service/Product/ReplicaManager.php

```diff
@@ -354,21 +355,74 @@ protected function getBareIndexNameFromReplicaSetting(string $replicaSetting): s
     }
 
     /**
-     * @param array $replicasToDelete
-     * @param bool $waitLastTask
+     * Delete replica indices
+     *
+     * @param array $replicasToDelete - which replicas to delete
+     * @param bool $waitLastTask - wait until deleting next replica (default: false)
+     * @param bool $prevalidate - verify replica is not attached to a primary index before attempting to delete (default: false)
      * @return void
      * @throws AlgoliaException
+     * @throws ExceededRetriesException
      */
-    protected function deleteIndices(array $replicasToDelete, bool $waitLastTask = false): void
+    protected function deleteReplicas(array $replicasToDelete, bool $waitLastTask = false, bool $prevalidate = false): void
     {
         foreach ($replicasToDelete as $deletedReplica) {
-            $this->algoliaHelper->deleteIndex($deletedReplica);
+            $this->deleteReplica($deletedReplica, $prevalidate);
             if ($waitLastTask) {
                 $this->algoliaHelper->waitLastTask($deletedReplica);
             }
         }
     }
 
+    /**
+     * @throws AlgoliaException
+     * @throws ExceededRetriesException
+     */
+    protected function deleteReplica(string $replicaIndexName, bool $precheck = false): void
+    {
+        if ($precheck) {
+            $settings = $this->algoliaHelper->getSettings($replicaIndexName);
+            if (isset($settings[self::ALGOLIA_SETTINGS_KEY_PRIMARY])) {
+                $this->detachReplica($settings[self::ALGOLIA_SETTINGS_KEY_PRIMARY], $replicaIndexName);
+            }
+        }
+
+        $this->algoliaHelper->deleteIndex($replicaIndexName);
+    }
+
+    /**
+     * Detach a single replica from its primary index
+     *
+     * @throws ExceededRetriesException
+     * @throws AlgoliaException
+     */
+    protected function detachReplica(string $primaryIndexName, string $replicaIndexName): void
+    {
+        $settings = $this->algoliaHelper->getSettings($primaryIndexName);
+        if (!isset($settings[self::ALGOLIA_SETTINGS_KEY_REPLICAS])) {
+            return;
+        }
+        $newReplicas = $this->removeReplicaFromReplicaSetting($settings[self::ALGOLIA_SETTINGS_KEY_REPLICAS], $replicaIndexName);
+        $this->algoliaHelper->setSettings($primaryIndexName, [ self::ALGOLIA_SETTINGS_KEY_REPLICAS => $newReplicas]);
```

Comment:
Is there a way to perform only one `setSettings`/`waitLastTask` in the process and deleting all replicas all at once ?